app-backup/luckybackup/luckybackup-0.4.9.recipe

```diff
@@ -0,0 +1,97 @@
+SUMMARY="A powerful, fast and reliable backup & sync tool"
+DESCRIPTION="An application that backs-up and/or synchronizes any directories \
+with the power of rsync. It is simple to use, fast (transfers over only \
+changes made and not all data), safe (keeps your data safe by checking all \
+declared directories before proceeding in any data manipulation), reliable and \
+fully customizable."
+COPYRIGHT="2008-2017 Loukas Avgeriou"
+LICENSE="GNU GPL v3"
+REVISION="1"
+HOMEPAGE="http://luckybackup.sourceforge.net"
+SOURCE_URI="https://downloads.sourceforge.net/luckybackup/luckybackup-$portVersion.tar.gz"
+CHECKSUM_SHA256="c161e98a11de5ad0a32a8e060e3fd45695a93ea3adf0581764a6b5d32e8da23b"
+PATCHES="luckybackup-$portVersion.patchset"
+ADDITIONAL_FILES="luckybackup.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+SECONDARY_ARCHITECTURES="x86"
+fi
+
+PROVIDES="
+	luckybackup$secondaryArchSuffix = $portVersion
+	app:LuckyBackup = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Network$secondaryArchSuffix
+	lib:libQt5WebKit$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:librsync$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libGL$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Network$secondaryArchSuffix
+	devel:libQt5WebKit$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:librsync$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qmake$secondaryArchSuffix >= 5
+	"
+
+BUILD()
+{
+	qmake luckybackup.pro
+	make $jobArgs
+}
+
+INSTALL()
+{
+	#create directories
+	mkdir -p "$appsDir"/LuckyBackup/{manual,translations,license}
+	#install
+	install -T ./luckybackup "$appsDir"/LuckyBackup/LuckyBackup
+
+	# copy manpages
+	mkdir -p "$manDir"/man8
+	cp -r ./manpage/*.8 "$manDir"/man8/
+
+	# copy html documentation
+	cp -r ./manual/* "$appsDir"/LuckyBackup/manual/
+	# copy translation
+	cp -rd ./translations/*.qm "$appsDir"/LuckyBackup/translations/
+	# copy license
+	cp -rd ./license/* "$appsDir"/LuckyBackup/license/
```

Comment:
I was if we could use `make install` and it turns out that we can do that if we patch the `luckybackup.pro`. I tried something and it allows to simplify `INSTALL` and switch to the layout I was suggesting... but the bad news is that the issue about the license not being displayed (`Could not locate the license file!`) is back...
And the "Help" > "LuckyBackup Handbook" menu displays an error:
```
ERROR
Could not locate the file: manual/index.html
```
(I did not check if it was OK without my changes.)
Anyway, this would require more digging...
So I don't know if this is worth the trouble.
Maybe we should just investigate about the settings file.
```diff
 	cmd:qmake$secondaryArchSuffix >= 5
 	"
 
+PATCH()
+{
+	sed -i \
+		-e "s|^\(documentation\.path =\).*|\1 $docDir|" \
+		-e "s|^\(translations\.path =\).*|\1 $dataDir/luckybackup|" \
+		-e "s|^\(manpage\.path =\).*|\1 $manDir/man8|" \
+		-e "s|^\(license\.path =\).*|\1 $docDir|" \
+		luckybackup.pro
+}
+
 BUILD()
 {
 	qmake luckybackup.pro
 	make $jobArgs
 }
 
 INSTALL()
 {
-	#create directories
-	mkdir -p "$appsDir"/LuckyBackup/{manual,translations,license}
-	#install
-	install -T ./luckybackup "$appsDir"/LuckyBackup/LuckyBackup
-
-	# copy manpages
-	mkdir -p "$manDir"/man8
-	cp -r ./manpage/*.8 "$manDir"/man8/
-
-	# copy html documentation
-	cp -r ./manual/* "$appsDir"/LuckyBackup/manual/
-	# copy translation
-	cp -rd ./translations/*.qm "$appsDir"/LuckyBackup/translations/
-	# copy license
-	cp -rd ./license/* "$appsDir"/LuckyBackup/license/
+	make install
+	install -d "$appsDir"
+	install -T luckybackup "$appsDir"/LuckyBackup
+	gunzip "$manDir"/man8/luckybackup.8.gz
 
 	local APP_SIGNATURE="application/x-vnd.LuckyBackup"
 	local MAJOR="`echo "$portVersion" | cut -d. -f1`"
 	local MIDDLE="`echo "$portVersion" | cut -d. -f2`"
 	local MINOR="`echo "$portVersion" | cut -d. -f3`"
 	local LONG_INFO="$SUMMARY"
 	sed \
 		-e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
 		-e "s|@MAJOR@|$MAJOR|" \
 		-e "s|@MIDDLE@|$MIDDLE|" \
 		-e "s|@MINOR@|$MINOR|" \
 		-e "s|@LONG_INFO@|$LONG_INFO|" \
 		"$portDir"/additional-files/luckybackup.rdef.in > luckybackup.rdef
 
-	addResourcesToBinaries luckybackup.rdef "$appsDir"/LuckyBackup/LuckyBackup
-	addAppDeskbarSymlink "$appsDir"/LuckyBackup/LuckyBackup
+	addResourcesToBinaries luckybackup.rdef "$appsDir"/LuckyBackup
+	addAppDeskbarSymlink "$appsDir"/LuckyBackup
 }
```

Reply:
Maybe change:
* `apps/LuckyBackup/license/`:arrow_right: `data/luckybackup/`
into ?
* `apps/LuckyBackup/license/`:arrow_right: `data/luckybackup/license/`
Also with sed you used $docDir?

Reply:
`src/global.cpp` had everything we needed to fix most issues. :smile: